### PR TITLE
Add service and endpoints and remove legacy resources from framework

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#2023

- Completes the endpoints resource by adding update and delete support.
- Wires the service and endpoints resources into the "cloud formation" bundle.
- Removes the legacy resource from the "cloud formation" bundle because all resources are migrated.

### CODE REVIEW

File: service/framework.go [hunk 1]
```diff
@@ -335,17 +364,16 @@ func newCRDFramework(config Config) (*framework.Framework, error) {
...
-	// TODO Remove the legacy resource once all resources are migrated to
-	// Cloud Formation.
 	var resources []framework.Resource
 	{
 		resources = []framework.Resource{
-			namespaceResource,
 			kmsKeyResource,
 			s3BucketResource,
 			s3BucketObjectResource,
-			legacyResource,
```
Comment (rossf7) [inline service/framework.go]: There goes legacy! 🎉 

Shame we can't delete it quite yet!

File: service/resource/servicev2/delete_test.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func Test_Resource_Service_newDeleteChange(t *testing.T) {
 		expectedService *apiv1.Service
 	}{
 		{
-			description: "current service matches desired service, return desired service",
+			description: "current state matches desired state, return desired state",
```
Comment (rossf7) [inline service/resource/servicev2/delete_test.go]: Some tidying up of the service test.

File: service/resource/endpointsv2/update.go [hunk 3]
```diff
@@ -30,7 +56,29 @@ func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
...
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "finding out if the endpoints has to be updated")
+
+	var endpointsToUpdate *apiv1.Endpoints
+
+	// The subsets can change if the private IP of the master node has changed.
+	// We then need to update the endpoints resource.
```
Comment (rossf7) [inline service/resource/endpointsv2/update.go]: This is the check for if the master IP has changed.

File: service/resource/endpointsv2/delete_test.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		resourceConfig.K8sClient = fake.NewSimpleClientset()
+		resourceConfig.Logger = microloggertest.New()
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	for _, tc := range testCases {
+		result, err := newResource.newDeleteChange(context.TODO(), tc.obj, tc.cur, tc.des)
```
Comment (fgimenez) [inline service/resource/endpointsv2/delete_test.go]: Could we add subtests here?

File: service/resource/endpointsv2/update_test.go [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+		resourceConfig.K8sClient = fake.NewSimpleClientset()
+		resourceConfig.Logger = microloggertest.New()
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	for _, tc := range testCases {
+		result, err := newResource.newUpdateChange(context.TODO(), tc.obj, tc.cur, tc.des)
```
Comment (fgimenez) [inline service/resource/endpointsv2/update_test.go]: Same here, subtests would be nice.

**Pull request merged**